### PR TITLE
Fix instructions for cloning repository.

### PR DESCRIPTION
**Pull request by materiari**
PT story #108671334
We shouldn't have students fork the main repo, just clone it and do their work locally.

### CODE REVIEW

**Comment by mverdicchio** [comment]
I'm not sure -- if students don't fork, where do they commit their own changes since they don't have write access to the repo?

**Comment by materiari** [comment]
As they did in the other assignments, they commit locally or backup to other clones.  We've *never* had them fork because by committing they put their completed homework in a public repo!  Look at the instructions for all the previous homework.

**Comment by armandofox** [comment]
i'm of two minds about this and would appreciate colleague advice.

on the one hand ari is right, fork + push implies most students will end up pushing their solution to a public repo.

on the other hand, cloning and then "scrubbing" the repo and backing up locally is just an unnatural workflow for github.  and if even one instructor tells their students to fork + push, there will be *some* set of reference solutions publicly available, and github is easy to search...

@tansaku has long argued for not caring about student homework solutions being public, and indeed for making the reference solutions public from day one.  i know that plagiarism can't be prevented by trying to keep reference solutions private, but i also don't want to hand the cheating opportunity to the students directly.  (i don't think most students cheat, but as instructors we know that the few who do can complicate things for everyone else.)

for my Spring offering of the course, i was thinking of doing some amateur steganography, putting a few "hidden markers" into the reference solutions so that students who try to clone them wholesale would get caught, and then making the reference solutions public.  any reactions?

**Comment by mverdicchio** [comment]
I'm more comfortable with Ari's position now that we promote a cloud based
IDE that makes pushing to github less critical than with the VM that could
become corrupted at any time.

But I side more with Sam and Armando in that the code is already out there
and someone's going to find it if they want to.

I like the idea of hiding stuff in there to prevent blatant copying, but I
suppose that only helps in the case of us providing reference solutions,
and not from the original starter code.

It's not unreasonable to have students commit 'locally' to a cloud repo or
push to a private repo. But the fact is that many students are brand new to
git and also may not have private repos available. Github education account
requests are often delayed and/or rejected.

tl;dr I vote for fork and clone and I'll probably tell my students to do it
that way unless there's a specific request not to.
Best,
Mike

**Comment by materiari** [comment]
Another possibility is to use clean and smudge filters in git to leave a few little footprints in the repo.  We know the users at least have ruby around, right?  Could an advanced user get around it?  _Of course._  But someone naively copying someone else's repo would leave a rather embarrassing paper trail.
https://git-scm.com/book/en/v2/Customizing-Git-Git-Attributes#Keyword-Expansion

**Comment by tansaku** [comment]
I say have everyone submit videos of them completing the homework as evidence that they did the coding themselves ... It's pretty trivial to provide that evidence these days and is a strong discourager of cheating  - that's what I always do in my HPU classes - have all students submit videos of themselves pairing or soloing on assignments

**Comment by armandofox** [comment]
I think we have lost the battle about public repos, so I'm going to support @tansaku and prefer workflows that resemble reality. For better or worse we need to be moving to a model where these homework assignments themselves don't count for much in terms of assessment. FWIW the way we're doing it at Berkeley is having exam questions that lean heavily on having done the HWs, including coding questions.
